File: backend/models/card.go
package models

// Card represents a Yu-Gi-Oh card from The Sacred Cards
type Card struct {
	ID           int    `json:"id" db:"id"`
	Name         string `json:"name" db:"name"`
	Description  string `json:"description" db:"description"`
	Image        string `json:"image" db:"image"`
	Type         string `json:"type" db:"type"`         // Monster, Spell, Trap
	Attribute    string `json:"attribute,omitempty" db:"attribute"` // Dark, Light, Earth, Water, Fire, Wind, Divine
	Race         string `json:"race,omitempty" db:"race"`           // Dragon, Spellcaster, Warrior, etc.
	Level        int    `json:"level" db:"level"`       // Monster level (0 for Spell/Trap)
	AttackPoints int    `json:"attack_points" db:"attack_points"`
	DefensePoints int   `json:"defense_points" db:"defense_points"`
	Cost         int    `json:"cost" db:"cost"`
	Rarity       string `json:"rarity,omitempty" db:"rarity"` // Common, Rare, Super Rare, Ultra Rare
}

// DeckSummary represents a deck summary for listing (without full card details)
type DeckSummary struct {
	ID            int    `json:"id" db:"id"`
	Name          string `json:"name" db:"name"`
	Description   string `json:"description" db:"description"`
	CharacterName string `json:"character_name,omitempty" db:"character_name"` // Character who uses this deck
	Archetype     string `json:"archetype,omitempty" db:"archetype"`            // Deck archetype/style
	MaxCost       int    `json:"max_cost" db:"max_cost"`
	TotalCost     int    `json:"total_cost"`
	CardCount     int    `json:"card_count"`
	IsPreset      bool   `json:"is_preset,omitempty" db:"is_preset"`
}

// DeckWithCards represents a deck with full card details
type DeckWithCards struct {
	ID            int    `json:"id" db:"id"`
	Name          string `json:"name" db:"name"`
	Description   string `json:"description" db:"description"`
	CharacterName string `json:"character_name,omitempty" db:"character_name"` // Character who uses this deck
	Archetype     string `json:"archetype,omitempty" db:"archetype"`            // Deck archetype/style
	Cards         []Card `json:"cards"`
	MaxCost       int    `json:"max_cost" db:"max_cost"`
	TotalCost     int    `json:"total_cost"`
	IsPreset      bool   `json:"is_preset,omitempty" db:"is_preset"`
}

File: backend/src/main/java/com/yugioh/model/Card.java
package com.yugioh.model;

import jakarta.persistence.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "cards")
public class Card {
    @Id
    private Integer id;

    @Column(nullable = false)
    private String name;

    @Column(columnDefinition = "TEXT")
    private String description;

    @Column(length = 500)
    private String image;

    @Column(nullable = false, length = 50)
    private String type; // Monster, Spell, Trap

    @Column(length = 50)
    private String attribute; // Dark, Light, Earth, Water, Fire, Wind, Divine

    @Column(length = 50)
    private String race; // Dragon, Spellcaster, Warrior, etc.

    @Column(nullable = false)
    private Integer level = 0; // Monster level (0 for Spell/Trap)

    @Column(name = "attack_points", nullable = false)
    private Integer attackPoints = 0;

    @Column(name = "defense_points", nullable = false)
    private Integer defensePoints = 0;

    @Column(nullable = false)
    private Integer cost = 0;

    @Column(length = 50)
    private String rarity; // Common, Rare, Super Rare, Ultra Rare

    @Column(name = "created_at")
    private LocalDateTime createdAt;

    @Column(name = "updated_at")
    private LocalDateTime updatedAt;

    // Getters and Setters
    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getImage() {
        return image;
    }

    public void setImage(String image) {
        this.image = image;
    }

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    public String getAttribute() {
        return attribute;
    }

    public void setAttribute(String attribute) {
        this.attribute = attribute;
    }

    public String getRace() {
        return race;
    }

    public void setRace(String race) {
        this.race = race;
    }

    public Integer getLevel() {
        return level;
    }

    public void setLevel(Integer level) {
        this.level = level;
    }

    public Integer getAttackPoints() {
        return attackPoints;
    }

    public void setAttackPoints(Integer attackPoints) {
        this.attackPoints = attackPoints;
    }

    public Integer getDefensePoints() {
        return defensePoints;
    }

    public void setDefensePoints(Integer defensePoints) {
        this.defensePoints = defensePoints;
    }

    public Integer getCost() {
        return cost;
    }

    public void setCost(Integer cost) {
        this.cost = cost;
    }

    public String getRarity() {
        return rarity;
    }

    public void setRarity(String rarity) {
        this.rarity = rarity;
    }

    public LocalDateTime getCreatedAt() {
        return createdAt;
    }

    public void setCreatedAt(LocalDateTime createdAt) {
        this.createdAt = createdAt;
    }

    public LocalDateTime getUpdatedAt() {
        return updatedAt;
    }

    public void setUpdatedAt(LocalDateTime updatedAt) {
        this.updatedAt = updatedAt;
    }
}
